### DISCUSSION TITLE
Accessing Task Instance Logs from callback hooks

### DISCUSSION THREAD

**Discussion by brunopastor** [apache/airflow#42716] [Q&A] [answered]
I need to access task instance logs from an `on_failure_callback` function. I understand that due to ORM's Session that's not possible since the session doesn't persist between task execution and callback hooks.

I tried the following without luck, getting the following output for `logs_data`:
```
task_logs_reader = TaskLogReader()
logs_data, _ = task_logs_reader.read_log_chunks(task_instance, task_instance.try_number, {})
```
`Could not read served logs: Parent instance  is not bound to a Session; lazy load operation of attribute 'trigger' cannot proceed (Background on this error at: https://sqlalche.me/e/14/bhk3)`

Is there any way to access task instance logs on `on_failure_callback`?

**Comment by potiuk** [comment]
You can use REST API.

**Comment by potiuk** [comment, marked as answer]
https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_log

**Reply by brunopastor** [reply]
Thank you very much! I can confirm it works!

**Reply by gulbvdvoort** [reply]
@brunopastor Can you show how you do implemented it?

**Reply by aglebov** [reply]
It somehow feels wrong that the callback code running inside Airflow has to use the REST API  and go through the trouble of authenticating itself to the same application.
